hal/src/nRF52840/ble_hal.cpp

```diff
@@ -1485,7 +1490,7 @@ int BleObject::Observer::processAdvReportEventFromThread(const ble_evt_t* event)
         // Scan response data packet
         hal_ble_scan_result_evt_t* result = getPendingResult(newAddr);
         if (!result) {
-            goto continue_scanning;
+            goto free;
```

Comment:
Can `continue_scanning` and `free` be simply coalesced, since there is anyway a check for `p_data`?

Reply:
No, we can't. For the case that an advertising packet is received and it expects a scan response packet, though the `p_data` is not `nullptr`, we cannot free this piece of memory, but continue scanning. This piece of memory will be freed when the scan response packet is received and notified to user application or the scan procedure finished.